pyccel/parser/semantic.py

```diff
@@ -631,9 +623,31 @@ def _infere_type(self, expr, **settings):
 
             return d_var
 
-        elif isinstance(expr, IndexedVariable):
+        elif isinstance(expr, Dlist):
```

Comment:
What is a `Dlist`?

Reply:
A Dlist is a list or a tuple multiplied by an integer. E.g:
`(1,False,4)*3`
`[1,2,3]*3`

Reply:
What about the concatenation of lists or tuples, do we have a special class for that too?

Reply:
The concatenation of lists or tuples appears to be broken

Reply:
@EmilyBourne Thanks for creating a new issue about it